source/common/upstream/health_checker_impl.cc

```diff
@@ -106,7 +106,7 @@ HttpHealthCheckerImpl::HttpHealthCheckerImpl(const Cluster& cluster,
                                           config.http_health_check().request_headers_to_remove())),
       http_status_checker_(config.http_health_check().expected_statuses(),
                            static_cast<uint64_t>(Http::Code::OK)),
-      codec_client_type_(codecClientType(config.http_health_check().use_http2())) {
```

Comment:
Where do you still support the deprecated option? This also needs deprecated feature tests.

Reply:
I tried to remove all the use_http2 usage. What is a deprecated feature test?

Reply:
You can't remove the existing support until it is fully deprecated. You need to translate use_http2 into the new thing in the interim, and have tests for that case. You can use DEPRECATED_FEATURE_TEST for that purpose.